src/Service/MethodRegistry.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace SilverStripe\MFA\Service;
+
+use LogicException;
+use SilverStripe\Core\Config\Configurable;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\MFA\Extension\MemberExtension;
+use SilverStripe\MFA\Method\MethodInterface;
+use SilverStripe\MFA\Model\RegisteredMethod;
+use SilverStripe\Security\Member;
+use UnexpectedValueException;
+
+/**
+ * A service class that holds the configuration for enabled MFA methods and facilitates providing these methods
+ */
+class MethodRegistry
+{
+    use Configurable;
+    use Injectable;
+
+    /**
+     * List of configured MFA methods. These should be class names that implement MethodInterface
+     *
+     * @config
+     * @var array
+     */
+    private static $methods;
+
+    /**
+     * Get implementations of all configured methods
+     *
+     * @return MethodInterface[]
+     */
+    public function getAllMethods()
```

Comment:
`getAllMethods()` versus `getMethods()` - any reason for the former?

Reply:
Nope. Yours is better. Updated